### PR TITLE
Add support for trusted authentication

### PR DESCRIPTION
**Pull request by KenKozman-GE**
This also includes the following:
- Remove the emsSystem id junk as that is always 1 in Azure.
- Get rid of a TODO around token race conditions.
- Switch to using DelegatingHandler vs. deriving from HttpClientHandler.
- Use MessageHandler stack to allow for some more invasive (but still black box) testing.
- Remove some other unused code.
- Add the ability to specify other settings per-call: e.g. correlation id.

### CODE REVIEW

**Comment by KenKozman-GE** [comment]
Note: This does NOT include the Polly stuff, but I want to look at that more next.

**Comment by c-owens** [comment]
I'll have to wait until tomorrow to give this a real review and get the tests running, but one thing I noticed right off the bat is we'll want to update `src/Directory.Build.props` and bump this to version 2.0.0 as well as update `ReleaseNotes.md` in the root.

**Comment by KenKozman-GE** [comment]
Ah dang, I totally missed that, will push it up in a sec.

**Comment by KenKozman-GE** [comment]
Thanks Cody I missed that client secret pass thing in the pipelines <facepalm>

File: .editorconfig [hunk 1]
```diff
@@ -236,32 +236,35 @@ dotnet_naming_style.first_upper.capitalization = first_word_upper
 # prefix_interface_interface_with_i - Interfaces must be PascalCase and the first character of an interface must be an 'I'
 dotnet_naming_style.prefix_interface_interface_with_i.capitalization = pascal_case
 dotnet_naming_style.prefix_interface_interface_with_i.required_prefix = I
+# non_public_fields - Non-Public fields should be camel case and prefixed with m_
+dotnet_naming_style.non_public_fields.capitalization = camel_case
```
Comment (c-owens) [inline .editorconfig:240]: Thanks for fixing these, I think this was the first editorconfig I tried to set up for c# and it had my head spinning :)

File: src/Client/V2/Service/MessageHandler.cs [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+                    deAuth = true;
+                }
+                if( value.TrustedAuthName != m_trustedAuthName )
+                {
+                    m_trustedAuthName = value.TrustedAuthName;
+                    deAuth = true;
+                }
+
+                if( deAuth )
+                    InvalidateAuthentication();
```
Comment (c-owens) [inline src/Client/V2/Service/MessageHandler.cs:78]: I think you can replace this call with `ClearAuthenticationCache()` instead and delete `InvalidateAuthentication`.

File: src/Tests/AuthenticationTrustedTests.cs [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        public void Trusted_info_but_no_client_id()
+        {
+            EmsApiServiceConfiguration config = m_config.Clone();
+            config.ApiClientId = null;
+            config.ApiClientSecret = null;
+            using var service = new EmsApiService( config );
+            var ctx = new CallContext
+            {
+                TrustedAuthName = "SAMAccountName",
+                TrustedAuthValue = "ksk"
```
Comment (c-owens) [inline src/Tests/AuthenticationTrustedTests.cs:54]: I have a service user in the US region w/ the name `emsapitest`, we should use that for the tests that will make a real call

**Review by c-owens — CHANGES_REQUESTED** [review]
I put some thought into your question about where we specify the handler chain, I think it may be a big pain to refactor things to get it into the service config itself, so I like the other option you mentioned (adding an optional `ApiServiceHttpClientConfig` in the service ctor).

You are also welcome to do nothing here and leave that as-is. I'd like to make one other breaking change and remove the default API endpoints since those are no longer valid, if you like I could tackle those two at the same time.

I am also curious if you're in a hurry to get this released and unblocked or if the release can wait a few days so I can make that endpoint change next week?

File: ReleaseNotes.md [hunk 4]
```diff
@@ -1,2 +1,12 @@
+# 2.0.0
```
[the remark this answers is not in the document]
Comment (KenKozman-GE) [reply ReleaseNotes.md:1]: Oh my bad... so basically no formatting?

[the remark this answers is not in the document]
Comment (KenKozman-GE) [reply ReleaseNotes.md:1]: Done.

File: src/Client/V2/Service/MessageHandler.cs (hunk 2 above)
Comment (KenKozman-GE) [reply src/Client/V2/Service/MessageHandler.cs:78]: Good catch!

File: src/Tests/AuthenticationTrustedTests.cs (hunk 3 above)
Comment (KenKozman-GE) [reply src/Tests/AuthenticationTrustedTests.cs]: Good call. We need two users, Cody and I decide on `emsapitest` and `emsweb` as those should stick around.

**Review by c-owens — APPROVED** [review]

**Pull request merged**